src/interface/src/app/types/plan.types.ts

```diff
@@ -14,6 +14,8 @@ export interface Plan extends BasePlan {
   area_acres: number;
   area_m2: number;
   creator: string;
+  role?: string;
+  permissions?: string[];
```

Comment:
just curious: are you just explicitly including all of these properties for readability? I know you want to tidy up this whole thing.